src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java

```diff
@@ -849,18 +849,23 @@ public final void markDirty()
      * register a block and position.
      *
      * @param block to be registered
-     * @param pos of the block
+     * @param pos   of the block
+     * @param world the world to place the tileentity in
      */
-    public void registerBlockPosition(@NotNull Block block, @NotNull final BlockPos pos)
+    public void registerBlockPosition(@NotNull Block block, @NotNull final BlockPos pos, @NotNull final World world)
     {
-        if (block instanceof BlockContainer)
+
+        final TileEntity tile = world.getTileEntity(pos);
+
+        if (tile != null && tile.hasCapability(CapabilityItemHandler.ITEM_HANDLER_CAPABILITY, null))
```

Comment:
Please do not do this. Use the inventoryutils method to get all itemhandlers in a te, then check for an empty list. This will break on a sided block.